service/create/service_test.go

```diff
@@ -0,0 +1,239 @@
+				Reservations: []*ec2.Reservation{
+					{
+						Instances: []*ec2.Instance{
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2RunningState)),
+								},
+							},
+						},
+					},
+				},
+			},
+			state: EC2RunningState,
+			res:   true,
+		},
+		{
+			name: "Expect terminated with two terminated instances in different reservations",
+			instances: &ec2.DescribeInstancesOutput{
+				Reservations: []*ec2.Reservation{
+					{
+						Instances: []*ec2.Instance{
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+						},
+					},
+					{
+						Instances: []*ec2.Instance{
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+						},
+					},
+				},
+			},
+			state: EC2TerminatedState,
+			res:   true,
+		},
+		{
+			name: "Expect not terminated with two terminated instances and one stopping instance in different reservations",
+			instances: &ec2.DescribeInstancesOutput{
+				Reservations: []*ec2.Reservation{
+					{
+						Instances: []*ec2.Instance{
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+						},
+					},
+					{
+						Instances: []*ec2.Instance{
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2StoppingState)),
+								},
+							},
+						},
+					},
+					{
+						Instances: []*ec2.Instance{
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+						},
+					},
+				},
+			},
+			state: EC2TerminatedState,
+			res:   false,
+		},
+		{
+			name: "Expect not terminated with two terminated instances in one reservation, one terminated and one stopping in another reservation, and one terminated in another reservation",
+			instances: &ec2.DescribeInstancesOutput{
+				Reservations: []*ec2.Reservation{
+					{
+						Instances: []*ec2.Instance{
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+						},
+					},
+					{
+						Instances: []*ec2.Instance{
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2StoppingState)),
+								},
+							},
+						},
+					},
+					{
+						Instances: []*ec2.Instance{
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+						},
+					},
+				},
+			},
+			state: EC2TerminatedState,
+			res:   false,
+		},
+	}
+
+	for i, tc := range tests {
+		res := allExistingInstancesMatch(tc.instances, tc.state)
+		if res != tc.res {
```

Comment:
Could you use https://github.com/stretchr/testify? We already have it vendored and I used it for k8s client testing.